runtime/near-evm-runner/src/lib.rs

```diff
@@ -166,8 +183,28 @@ impl<'a> EvmContext<'a> {
         self.add_balance(&sender, U256::from(self.attached_deposit))?;
         let value =
             if self.attached_deposit == 0 { None } else { Some(U256::from(self.attached_deposit)) };
-        interpreter::call(self, &sender, &sender, value, 0, &contract_address, &input, true)
-            .map(|rd| rd.to_vec())
+        let rd = interpreter::call(
+            self,
+            &sender,
+            &sender,
+            value,
+            0,
+            &contract_address,
+            &input,
+            true,
+            &self.evm_gas_counter.gas_left(),
+        )?;
+        match rd {
+            MessageCallResult::Success(gas_left, data) => {
```

Comment:
You mean `impl From<MessageCallResult> for Result<Vec>`? will give a try

Reply:
Seems not work because we need to get the gas_left, and call self.evm_gas_counter to set gas_left. 
```
        match rd {
            MessageCallResult::Success(gas_left, data) => {
                self.evm_gas_counter.set_gas_left(gas_left);
                Ok(data.to_vec())
            }
```
but impl From<MessageCallResult> doesn't have access to self.evm_gas_counter, so we'll need to match rd anyway